### ISSUE TITLE
How do I get the number of pages indexed with their respective date?

### ISSUE THREAD

**Issue by luizcarlospedrosogomes** [google/google-api-php-client#1364]
I can get the prints with: 
```
foreach($this->getSites()->searchanalytics->query($_POST['site'], $opcao)->getRows() as $impre){
                        if($impre['impressions'] > 0){
                            array_push($impressoes,$impre['impressions']);
                            foreach($impre['keys'] as $keys)
                                array_push($periodo, date("d/m/Y", strtotime($keys)));
                        }
                    }

```
But when I try to pick it up, using code
$sitemaps->listSitemaps("http://www.overer.com/")

it returns me only the last date. I would like to be able to filter by the start and end date with a return of the date and number of pages indexed on the respective date.

Does api support this?
is there any example of how to do this?

**Comment by mattwhisenhunt** [comment]
There is not a PHP example for the Google Search Console API but you can adapt [the python one](https://developers.google.com/webmaster-tools/search-console-api-original/v3/quickstart/quickstart-python) and [a PHP example from another API](https://github.com/google/google-api-php-client/tree/master/examples) and get up to speed pretty quickly.

Here are [the docs](https://developers.google.com/webmaster-tools/search-console-api-original/v3/parameters) and I find the [API Explorer](https://developers.google.com/apis-explorer/#search/webmasters/webmasters/v3/) to be an invaluable resource as well.

**Comment by luizcarlospedrosogomes** [comment]
using the docs i get number impressions for date, i do use the docs. for get impressios i see example python.

for indexed i use this code:

> debug($sitemaps->listSitemaps("http://www.dimba.com.br/"));

but return :

```
object(Google_Service_Webmasters_SitemapsListResponse) {
	[protected] collection_key => 'sitemap'
	[protected] internal_gapi_mappings => array()
	[protected] sitemapType => 'Google_Service_Webmasters_WmxSitemap'
	[protected] sitemapDataType => 'array'
	[protected] modelData => array(
		'sitemap' => array(
			(int) 0 => array(
				'path' => 'http://www.dominui.com.br/sitemap.xml',
				'lastSubmitted' => '2017-12-31T17:06:56.592Z',
				'isPending' => false,
				'isSitemapsIndex' => false,
				'type' => 'sitemap',
				'lastDownloaded' => '2018-01-03T11:42:39.262Z',
				'warnings' => '0',
				'errors' => '0',
				'contents' => array(
					(int) 0 => array(
						'type' => 'web',
						'submitted' => '4',
						'indexed' => '0'
					)
				)
			)
		)
	)
	[protected] processed => array()
}
```


I need the indexed page number to be associated with a date.

I also need the date range to be sent as a parameter.

just as it is in
https://www.google.com/webmasters/tools/index-status?hl=pt-BR&siteUrl={mydominio}
![image](https://user-images.githubusercontent.com/9791364/34655437-ebffa62c-f3f0-11e7-9cea-0ba749248e3f.png)



basically it is exactly as it is in the graph of indexed pages of google.

I already read the documentation and found nothing.

I also tried crawling directly on the google page, but with no success.

Any more ideas? will it be that google api does not let you get the graph of indexed pages?

**Comment by mattwhisenhunt** [comment]
I would get your query working in API Explorer or the 'Try this API' feature of [REFERÊNCIA](https://developers.google.com/webmaster-tools/search-console-api-original/v3/searchanalytics/query) with your date range. That may give you an insight in to which methods or parameters you need to use from the services classes.

Those cURL and Javascript tabs are new to me, and depending on your comfort level either PHP's libcurl module or frontend Javascript you maybe get some mileage out of using the code generated in the full page mode of _Try this API_.

**Comment by luizcarlospedrosogomes** [comment]
Basically the api documentation does not say there is a return for indexing. just for click, position, ctr and print.

I am deducing this simple functionality does not exist in google api.


Is that right?

**Comment by mattwhisenhunt** [comment]
I don't have any experience with that API, but I did read something about using custom fields yesterday. It maybe that the API is intentionally simple with the expectation most people with want to extend it in a unknown way.

I think in general that is right though, you tend to have to download all entries from an API and store them in a cache or database and then only filter them in an interesting way.

**Comment by luizcarlospedrosogomes** [comment]
Thanks for the answers.

is there a way to get the index status graph?

I would have an example, it could be in another API or language, for me to adapt.

thank you

**Comment by mattwhisenhunt** [comment]
I don't know how to do that and I am not going to be able to figure it out in the foreseeable future. You might want to search or ask on StackOverflow. I will leave this issue open for a few weeks in case anyone else sees it and knows the answer.

**Comment by luizcarlospedrosogomes** [comment]
https://webmasters.googleblog.com/2018/01/introducing-new-search-console.html
